test/utils.cpp

```diff
@@ -147,6 +147,38 @@ TEST_CASE("tokenize_quoted() doesn't un-escape escaped backticks", "[utils]")
 	REQUIRE(tokens[1] == "\\`foobar `bla`\\`");
 }
 
+TEST_CASE("tokenize_nl()","[utils]")
+{
+	std::vector<std::string> tokens;
+
+	SECTION("correctly tokenizes a string","[utils]")
+	{
+		tokens = utils::tokenize_nl("first\nsecond\n\nthird");
+
+		REQUIRE(tokens[0] == "first");
```

Comment:
Please check the length of the vector as well, both here and further down below. That way, we can be sure that other `REQUIRE`s check the result of the function and not some random junk from memory.

Reply:
Thank you.

Reply:
Sorry, I meant a check *before* the other checks.

Right now, if `tokenize_nl` returns e.g. 1 element instead of 5, this test will read uninitialized memory in `tokens[2]`, and either fail the assertion or segfault. Both scenarios are bad, because:
1. failed assertion will make the other programmer think that they return random junk from the function;
2. segfault will have to be debugged, which takes time.

Both problems can be prevented if the length is checked upfront.

Reply:
Makes sense. Done